mne/viz/tests/test_3d.py

```diff
@@ -302,4 +302,40 @@ def test_snapshot_brain_montage():
     assert_raises(TypeError, snapshot_brain_montage, fig, info)
 
 
+@testing.requires_testing_data
+@requires_pysurfer
+@requires_mayavi
+def test_plot_vec_source_estimates():
+    """Test plotting of vector source estimates."""
+    mlab = _import_mlab()
+    sample_src = read_source_spaces(src_fname)
+
+    vertices = [s['vertno'] for s in sample_src]
+    n_verts = sum(len(v) for v in vertices)
+    n_time = 5
+    data = np.random.RandomState(0).rand(n_verts, 3, n_time)
+    stc = VectorSourceEstimate(data, vertices, 1, 1)
+
+    with warnings.catch_warnings(record=True):
+        warnings.simplefilter('always')
+        stc.plot('sample', subjects_dir=subjects_dir)
+
+        # Do multiple hemispheres in one view
+        stc.plot('sample', subjects_dir=subjects_dir, hemi='both')
+
+        # Do single hemispheres in multiple views
+        stc.plot('sample', subjects_dir=subjects_dir, hemi='lh',
+                 views=['lat', 'ven'])
+
+        # Do multiple hemispheres in multiple views
+        stc.plot('sample', subjects_dir=subjects_dir, hemi='both',
+                 views=['lat', 'ven'])
```

Comment:
These two tests can probably be removed since they're essentially testing ``plot_source_estimates`` which should be tested elsewhere. You can also squeeze off a couple of seconds by setting ``smoothing_steps=1`` in the other call to stc.plot.
Regarding the invalid pixmap error, I remember seeing it on Circleci, but I can't remember how it was fixed.